api/core/v2/silenced.go

```diff
@@ -84,6 +86,27 @@ func (s *Silenced) Prepare(ctx context.Context) {
 	}
 }
 
+// Matches returns true if the given check name and subscription match the silence.
+//
+// The two properties compared, Subscription and Check, are only compared if they are
+// not empty. An empty value for either of those fields is considered a wildcard,
+// ie: s.Check = "foo" && s.Subscription = "" will return true for s.Matches("foo", <anything>).
+func (s *Silenced) Matches(check, subscription string) bool {
+	if s == nil {
+		return false
+	}
+
+	if !stringsutil.InArray(s.Subscription, []string{"", "*", subscription}) {
```

Comment:
This is a bug fyi @palourde @echlebek, same in the condition below with `s.Check`.

I found this while doing some adhoc testing of the PR comments on https://github.com/sensu/sensu-go/pull/3382#issuecomment-550450740

The first condition in the implementation of `stringsutil.InArray` is:
https://github.com/sensu/sensu-go/blob/3860ab4/util/strings/strings.go#L15-L18

Which means empty strings for either subscription or check are not treated as wildcards as intended.

I will push a separate PR with these two conditions changed to address.

Reply:
Fix is here: https://github.com/sensu/sensu-go/pull/3387